aiohttp/locks.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+import collections
+
+from .helpers import ensure_future
+
+
+class ErrorfulOneShotEvent:
```

Comment:
The name is too long.
Also it starts with `Error`, usually this schema is used for nameing exception classes.